tracer/build/_build/Build.Utilities.cs

```diff
@@ -120,28 +120,11 @@ partial class Build
         .Requires(() => Framework)
         .Executes(() =>
         {
-            var envVars = new Dictionary<string,string>()
+            var envVars = new Dictionary<string, string>()
             {
-                {"COR_ENABLE_PROFILING", "1"},
-                {"COR_PROFILER", "{846F5F1C-F9AE-4B07-969E-05C26BC060D8}"},
-                {"COR_PROFILER_PATH_32", TracerHomeDirectory / "win-x86" / "Datadog.Trace.ClrProfiler.Native.dll"},
-                {"COR_PROFILER_PATH_64", TracerHomeDirectory / "win-x64" / "Datadog.Trace.ClrProfiler.Native.dll"},
-                {"CORECLR_ENABLE_PROFILING", "1"},
-                {"CORECLR_PROFILER", "{846F5F1C-F9AE-4B07-969E-05C26BC060D8}"},
-                {"DD_DOTNET_TRACER_HOME", TracerHomeDirectory },
-                {"ASPNETCORE_URLS", "http://*:5003" },
+                { "ASPNETCORE_URLS", "http://*:5003" }
             };
```

Comment:
This has changed behaviour slightly, and I find it slightly more confusing, as we're setting two lots of environment variables, some of which could be overrides etc.

Personally I'd prefer to have a `GetProfilerEnvironmentVariables()` method, which allows us to to still keep the same behaviour here, e.g. 

```csharp
public static Dictionary<string, string> GetProfilerEnvironmentVariables(AbsolutePath tracerHomeDirectory)
    {
        var envVars = new Dictionary<string, string>()
        {
            {"COR_ENABLE_PROFILING", "1"},
            {"COR_PROFILER", "{846F5F1C-F9AE-4B07-969E-05C26BC060D8}"},
            {"COR_PROFILER_PATH_32", tracerHomeDirectory / "win-x86" / "Datadog.Trace.ClrProfiler.Native.dll"},
            {"COR_PROFILER_PATH_64", tracerHomeDirectory / "win-x64" / "Datadog.Trace.ClrProfiler.Native.dll"},
            {"CORECLR_ENABLE_PROFILING", "1"},
            {"CORECLR_PROFILER", "{846F5F1C-F9AE-4B07-969E-05C26BC060D8}"},
            {"DD_INTEGRATIONS", tracerHomeDirectory / "integrations.json" },
            {"DD_DOTNET_TRACER_HOME", tracerHomeDirectory },
        };

        if (EnvironmentInfo.IsWin)
        {
            envVars.Add("CORECLR_PROFILER_PATH_32", tracerHomeDirectory / "win-x86" / "Datadog.Trace.ClrProfiler.Native.dll");
            envVars.Add("CORECLR_PROFILER_PATH_64", tracerHomeDirectory / "win-x64" / "Datadog.Trace.ClrProfiler.Native.dll");
        }
        else
        {
            envVars.Add("CORECLR_PROFILER_PATH", tracerHomeDirectory / "Datadog.Trace.ClrProfiler.Native.so");
        }

        return envVars;
    }
```

and then this becomes

```csharp
var envVars = GetProfilerEnvironmentVariables();
envVars["ASPNETCORE_URLS] = "http://*:5003";

if (ExtraEnvVars?.Length > 0)
// etc
```

Reply:
Like the idea. However, not private method, not `DotNetSettingsExtensions` class doesn't sounds like proper place for that.
Do we have somethings like "Constansts" class, where we can set all of those hard coded values in the same place?

Reply:
I expect so, but probably not in the build project. I don't see a problem with having it as a private method personally, as it's only going to be used by these two targets currently?🤷‍♂️